### PR TITLE
Stream Control

### PR DESCRIPTION
**Pull request by ndt93**
- Extend Wowza to support stopping streams using external HTTP requests
- Extend Wowza to verify streams with application server before a stream is allowed to published
- Implement adapter and REST API for stopping stream from the application server

### CODE REVIEW

**Comment by ndt93** [comment]
@sharminelow Can you help review this? Take notes of 2 places with `TODO`. We can arrange to merge your other PR first so I can help resolving conflicts with this if there are any.

File: app_server/app/services/StreamService.js [hunk 1]
```diff
@@ -72,6 +78,14 @@ Class.getListOfStreams = function(filters) {
   });
 };
 
+Class.stopStream = function(appName, appInstance, streamId) {
+  // TODO: Update stream's state in database before calling the below
```
Comment (sharminelow) [inline app_server/app/services/StreamService.js]: So the client will call `stopStream` and in this method we will call another method toggle the `live` status. Am I right to say that?

If that's the case, I will edit away some controller method on my branch. And you will need make a call to the `Service.endStream` which is a method to toggle the status.

Comment (ndt93) [inline app_server/app/services/StreamService.js]: Yep.

**Comment by sharminelow** [comment]
Checked through, shouldn't be a problem. You can merge yours in after the `TODO` has been resolved.

**Comment by ndt93** [comment]
Ok. Thanks :). I will review yours a while later since we need it to complete the TODO

**Pull request merged**